extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -151,14 +188,20 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
 
     this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
     this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
-    this.mutateElement(() => {
-      createShadowRootWithStyle(
-        this.element,
-        this.shoppingTagEl_,
-        shoppingTagCSS
-      );
-      this.hasAppendedInnerShoppingTagEl_ = true;
-    });
+
+    this.measureMutateElement(
+      () => {
+        createShadowRootWithStyle(
+          this.element,
+          this.shoppingTagEl_,
+          shoppingTagCSS
+        );
+        this.hasAppendedInnerShoppingTagEl_ = true;
+      },
+      () => {
+        this.styleTagText_();
```

Comment:
`styleTagText_()` already runs its own measure/mutateElement callback. Overall, it's usually not a good idea to run entire methods into a measure/mutateCallback because it becomes too hard to figure out what context you're in.

Reply:
Addressed in https://github.com/ampproject/amphtml/pull/37704